lens-server/src/main/java/org/apache/lens/server/query/LensServerDAO.java

```diff
@@ -821,4 +823,32 @@ public boolean deleteActiveSession(LensSessionHandle sessionId) throws LensExcep
 
     return result;
   }
+
+  /**
+   * DAO method to insert a new Prepared query into Table.
+   *
+   * @param preparedQueryContext to be inserted
+   * @throws SQLException the exception
+   */
+  public void insertPreparedQuery(PreparedQueryContext preparedQueryContext) throws SQLException {
```

Comment:
This is a void method, and anyways it calls external system to insert data, I don't think we should be unit-testing here.

Reply:
I am asking or unit test around new snippet of code written here and unit test is not only around return  value. 
Consider a scenario where in future any user wants to use this function and end up calling "insert prepare" twice from his code then this code is expected throw exception in the second call. A developer will not come to know this at build time unless unit tests are added. There can be many such scenarios unknown now.
Also please add checks and unit tests for invalid values of preparedQueryContext. It can cause runtime exception. rethrow LensException instead.

Reply:
Some one calling once/twice/zero times is not part of unit testing. It is a overkill to unnecessarily mock and unit test.
In this code snippet, it is making a call to a external system and it returns void.

Reply:
anyways I have added unit test case.